src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1762 @@
+                LogInfo.Log("Waking up device", LogLevel.Debug);
+                _controller.Write(_pin, PinValue.Low);
+                Thread.Sleep(4);
+                _controller.Write(_pin, PinValue.High);
+                byte[] samMessage = CreateWriteMessage(CommandSet.SAMConfiguration, new byte[3] { (byte)_securityAccessModuleMode, (byte)(_virtualCardTimeout), 0x00 });
+                byte[] wakeUp = new byte[I2cWakeUp.Length + samMessage.Length];
+                I2cWakeUp.CopyTo(wakeUp, 0);
+                samMessage.CopyTo(wakeUp, I2cWakeUp.Length);
+                _spiDevice.Write(wakeUp);
+
+            }
+            else if (_i2cDevice != null)
+            {
+                LogInfo.Log("Waking up PN522 on I2C mode", LogLevel.Debug);      
+                byte[] samMessage = CreateWriteMessage(CommandSet.SAMConfiguration, new byte[3] { (byte)_securityAccessModuleMode, (byte)(_virtualCardTimeout), 0x00 });
+                byte[] wakeUp = new byte[I2cWakeUp.Length + samMessage.Length];
+                I2cWakeUp.CopyTo(wakeUp, 0);
+                samMessage.CopyTo(wakeUp, I2cWakeUp.Length);
+                int i = 3;
+                while (i > 0)
+                {
+                    try
+                    {
+                        _i2cDevice.Write(wakeUp);
+                        LogInfo.Log($"Send: {BitConverter.ToString(wakeUp)}", LogLevel.Debug);
+                        break;
+                    }
+                    catch (IOException)
+                    {
+                        i--;
+                    }
+                }
+
+                byte[] buff = new byte[7];
+                do
+                {
+                    try
+                    {
+                        _i2cDevice.Read(buff);
+                    }
+                    catch (IOException)
+                    {
+                        throw new Exception("PN532 is not ready for I2C communication");
+                    }
+
+                }
+                while (buff[0] == 0x01);
+            }
+        }
+
+        #endregion
+
+        #region Communication
+
+        /// <summary>
+        /// Setup the baud rate communication when using the HSU Serial Port mode
+        /// </summary>
+        /// <param name="baudRate">Baud rate</param>
+        /// <returns>True if success</returns>
+        public bool SetSerialBaudRate(BaudRate baudRate)
+        {
+            if (_serialPort == null)
+                return false;
+            Span<byte> toSend = stackalloc byte[1] { (byte)baudRate };
+            var ret = WriteCommand(CommandSet.SetSerialBaudRate, toSend);
+            if (ret < 0) return false;
+            ret = ReadResponse(CommandSet.SetSerialBaudRate, Span<byte>.Empty);
+            // Need to send an acknowledge
+            _serialPort.Write(AckBuffer, 0, AckBuffer.Length);
+            _serialPort.Close();
+            switch (baudRate)
+            {
+                case BaudRate.B0009600:
```

Comment:
nit: you can call
`_serialPort.WriteTimeout = _serialPort.ReadTimeout;`
on the very bottom instead of that

also ReadTimeout seems to be very close to: `Math.Round(10252800.0 / BaudRate)` (equal or 1ms more in case of 230400 where you have to round 44.5)

Reply:
Yes for the math. I sticked to what is in the documentation page 39 at least, those are the official values. And added the comment on from where the data is coming from plus the approximation but left the data as they are so it's closer to the values in the documentation.